code/API_definitions/network_access_management.yaml

```diff
@@ -0,0 +1,2863 @@
+    - **API Client**: A first- or third-party software system using this API.
+
+    # API Functionality
+    This API allows API clients to manage isolated networks and reboot network access devices. OAuth 2.0 is used for authorizing an identity (resource holder) and scopes are used for role-based authorization.
+
+    ## Authorization
+    Authorization of API clients SHOULD be performed with OAuth 2.0 using the authorization code grant type such that service owners delegate access to their resources exposed by this API to the API client.  PKCE MUST be used when the API client secret cannot be secured (e.g., mobile application or single page application without a secure back-end).  This API MAY also be used with the client credentials grant type, though such a use case only makes sense for first-party network operator API clients or when the service owner operates their own API client (client ID is bound to the service owner's identity).
+
+    ### Scopes
+    OAuth 2.0 scopes are used to protect the service owner's information and privacy.  The general format is `<API_name>-<Resource>-<Permission>`.  A resource may be further broken down into `<Resource_name>-<Field>` or `<Resource_name>-<Role>` for granular permissions. Scopes also control output information as opposed to just authorizing endpoints. Each endpoint definition will clearly state what scopes are needed for access and which are needed to access additional information. Below are some general rules for scopes in this API.
+
+    - Regulated information, such as physical locations or hardware addresses of network devices, have special scopes.  Network operators are expected to follow any local regulations when sharing this information with an API client.
+    - Scopes have inheritance when one scope would otherwise require the inherited scope.
+    - Read scopes for one resource impacts the output of another resource when the requested resource has a relationship with the first (e.g., listing networks on a device requires both device and network read scopes).
+    - Isolated networks are broken down into 3 roles allowing service owners to grant granular permissions to API clients: only-own, primary, and auxiliary. The primary role allows API clients to manage the main network access configuration. The auxiliary role allows API clients to manage all auxiliary (i.e., secondary) networks. The only-own role allows API clients to manage networks it has created. Only-own networks SHOULD be an auxiliary, or secondary, network. Note that there is an edge case when a service owner revokes the primary role authorization. An API client SHOULD be able to continue managing primary isolated networks created by the client. Complete administrative control requires both the primary and auxiliary roles.
+
+    | Scope | Description | Inherits
+    |-------|-------------|---------
+    | **cpe-management-service-sites-read** | List a service owner's service sites

```

Comment:
As I said before, have a look at the CAMARA Commonalities guidelines for scope name format (it uses ":" instead of "-" among other things). But in any case, I guess that at least it would be `network-access-management-` instead of `cpe-management-`, right?

Reply:
Scopes are fitted into the updated format as best as possible.  Opened https://github.com/camaraproject/NetworkAccessManagement/issues/13 to follow up on fine-grained and role-based usage of scopes.